src/mean.ts

```diff
@@ -0,0 +1,87 @@
+import type {
+  IterableContainer,
+  NonEmptyArray,
+  NonEmptyReadonlyArray,
+} from "./internal/types";
+import { purry } from "./purry";
+import { sum } from "./sum";
+
+type Mean<T extends IterableContainer<bigint> | IterableContainer<number>> =
+  T extends NonEmptyArray<bigint> | NonEmptyReadonlyArray<bigint>
+    ? bigint
+    : T extends NonEmptyArray<number> | NonEmptyReadonlyArray<number>
+      ? number
+      : T extends [] | readonly []
```

Comment:
I agree, as the codebase increases in size, it is crucial to maintain consistency. While I haven't come across a lint rule specifically for enforcing consistency in type-level empty array checks, it might be worth exploring a custom one for this purpose?